### PR TITLE
[aboot]: refactor boot0.j2 to support one image SONiC-to-SONiC upgrade

### PR DESCRIPTION
**Pull request by lguohan**

### CODE REVIEW

File: files/Aboot/boot0.j2 [hunk 1]
```diff
@@ -120,6 +120,17 @@ platform_specific() {
     fi
 }
 
+# check the hash file in the image, and determine to install or just skip
+GIT_REVISION=$(unzip -p "$swipath" .imagehash)
+LOCAL_IMAGEHASH=$(cat $image_path/.imagehash 2>/dev/null || true)
```
Comment (qiluo-msft) [inline files/Aboot/boot0.j2]: > 2>/dev/null || true [](start = 44, length = 20)

No need to handle stderr and return code.
[https://www.gnu.org/software/bash/manual/bashref.html#Command-Substitution](https://www.gnu.org/software/bash/manual/bashref.html#Command-Substitution)

File: files/Aboot/boot0.j2 [hunk 2]
```diff
@@ -24,7 +24,7 @@ kernel_params=kernel-params
 
 aboot_machine="arista_unknown"
 
-target_path=/mnt/flash
+[ -z "$target_path" ] && target_path=/mnt/flash
```
Comment (qiluo-msft) [inline files/Aboot/boot0.j2]: Is it more readable to rewrite similar blocks by 'if-else'.
The shortcuts is useful for simple error handling, but confusing for complex conditions.

File: files/Aboot/boot0.j2 (hunk 1 above)
Comment (lguohan) [inline files/Aboot/boot0.j2]: it works. I can revisit it later, but I do not want to change this for this pr.

File: files/Aboot/boot0.j2 (hunk 2 above)
Comment (lguohan) [inline files/Aboot/boot0.j2]: if target_path is null, then assign target_path. It is the simplest logic.

**Pull request merged**